doc/Python-on-test-machines.md

```diff
@@ -0,0 +1,134 @@
+| test-osuosl-aix61-ppc64_be-3 | Python 3 | 2.7.11 | None | ...three
+| test-osuosl-centos7-ppc64_le-1 | Python 2 | 2.7.5 | 3.6.8 | https://github.com/nodejs/build/issues/1674
+| test-osuosl-centos7-ppc64_le-2 | Python 2 | 2.7.5 | 3.6.8 | https://github.com/nodejs/build/issues/1674
+| test-osuosl-ubuntu1404-ppc64_be-1 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-osuosl-ubuntu1404-ppc64_be-2 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-osuosl-ubuntu1404-ppc64_le-1 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-osuosl-ubuntu1404-ppc64_le-2 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-osuosl-ubuntu1404-ppc64_le-4 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-packetnet-centos7-arm64-1 | Py2 & Py3 | 2.7.5 | None | `yum search python3` shows Python 36
+| test-packetnet-centos7-arm64-2 | Py2 & Py3 | 2.7.5 | None | `yum search python3` shows Python 36
+| test-packetnet-ubuntu1604-arm64-1 | | 2.7.12 | 3.5.2 | 
+| test-packetnet-ubuntu1604-arm64-2 | | 2.7.12 | 3.5.2 | 
+| test-packetnet-ubuntu1604-x64-1 | | 2.7.12 | 3.5.2 | 
+| test-packetnet-ubuntu1604-x64-2 | | 2.7.12 | 3.5.2 | 
+| test-rackspace-centos7-x64-1 | Py2 & Py3 | 2.7.5 | None | `yum search python3` shows Python 36
+| test-rackspace-debian8-x64-1 | Python 3 | 2.7.9 | 3.4.3 | 
+| test-rackspace-debian8-x64-2 | Python 3| 2.7.9 | 3.4.2 | 
+| test-rackspace-fedora26-x64-1 | | 2.7.14 | 3.6.4 | 
+| test-rackspace-fedora27-x64-1 | | 2.7.14 | 3.6.4 | 
+| test-rackspace-freebsd10-x64-1 | | 2.7.15 | 3.6.7 | 
+| test-rackspace-ubuntu1204-x64-1 | Py2 & Py3 | 2.7.3 | None | `apt-cache search python3` shows Python 3.2 Debug build
+| test-rackspace-ubuntu1604-x64-1 | Python 3 | 2.7.12 | 3.5.2 | 
+| test-rackspace-ubuntu1604-x64-2 | Python 3 | 2.7.12 | 3.5.2 | 
+| test-scaleway-ubuntu1604-armv7l-1 | | 2.7.12 | 3.5.2 | 
+| test-scaleway-ubuntu1604-armv7l-2 | | 2.7.12 | 3.5.2 | 
+| test-scaleway-ubuntu1604-armv7l-3 | | 2.7.12 | 3.5.2 | 
+| test-softlayer-centos5-x64-1 | Py2 & Py3 | 2.4.3 | None | `yum search python3` does not have anything
+| test-softlayer-centos5-x64-2 | Py2 & Py3 | 2.4.3 | None | `yum search python3` does not have anything
+| test-softlayer-centos6-x64-1 | Py2 & Py3 | 2.6.6 | None | `yum search python3` shows Python 36
+| test-softlayer-centos6-x64-2 | Py2 & Py3 | 2.6.6 | None | `yum search python3` shows Python 36
+| test-softlayer-centos7-x64-1 | Python 3 | 2.7.5 | None | `yum search python3` shows Python 36
+| test-softlayer-debian8-x86-1 | Python 3 | 2.7.9 | 3.4.2 | 
+| test-softlayer-debian9-x64-1 | | 2.7.13 | 3.5.3 | 
+| test-softlayer-ubuntu1404-x64-1 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-softlayer-ubuntu1404-x86-1 | Py2 & Py3 | 2.7.6 | 3.4.3 | 
+| test-softlayer-ubuntu1604-x64-1 | | 2.7.12 | 3.5.2 | 
+| test-softlayer-ubuntu1604-docker-x64-1 | | 2.7.12 | 3.5.2 |
+
+
+### find_pythons.sh
+```sh
+#!/bin/sh
+
+echo "$0: $1"
+ssh $1 which python ; python --version || true
+ssh $1 which python2 ; python2 --version || true
+ssh $1 which python3 ; python3 --version || true
```

Comment:
`python3` is in `/opt/bin` on AIX, which is not in the default PATH. What do you suggest?

For compilers, we have a `select-compiler.sh` script, we could do something like that for Python. We could also extend the PATH manually with all possible locations, or perhaps have a `add-python3.sh` script that is called before python. Or, we could modify the config of all test systems to put `python3` in the default path. Could also symlink `python3` into the path (though I don't know if that is enough, it might depend on how Python resolves references to its `lib/`).

Reply:
export PATH=/opt/bin:$PATH